src/core/ext/xds/certificate_provider_factory.h

```diff
@@ -39,6 +39,8 @@ class CertificateProviderFactory {
     // Name of the type of the CertificateProvider. Unique to each type of
     // config.
     virtual const char* name() const = 0;
+
+    virtual std::string ToString() { return "{}"; }
```

Comment:
Why not make this pure virtual, so that implementations can't forget to implement it?

Reply:
just sheer laziness.. I'll update it